src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable max-len */
+import classNames from 'classnames';
+import { useState } from 'react';
+
+import { Todo } from '../../types/Todo';
+import { deleteTodo, updateTodo } from '../../api/todos';
+
+type Props = {
+  todo: Todo,
+  onTodosChange: React.Dispatch<React.SetStateAction<Todo[]>>,
+  onErrorMesssageChange: (val: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  onTodosChange,
+  onErrorMesssageChange,
+}) => {
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState(todo.title);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleTodoTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setEditedTitle(event.target.value);
+  };
+
+  const removeTodo = (todoId: number) => {
+    setIsLoading(true);
+
+    deleteTodo(todoId)
+      .then(() => onTodosChange(currentTodos => currentTodos
+        .filter(curTodo => curTodo.id !== todoId)))
+      .catch(() => onErrorMesssageChange('Unable to delete a todo'))
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleUpdate = (data: Partial<Todo>) => {
+    setIsLoading(true);
+
+    updateTodo(todo.id, data)
+      .then((newTodo) => {
+        onTodosChange(prevState => {
+          return prevState.map(todoItem => {
+            if (todoItem.id === newTodo.id) {
+              return newTodo;
+            }
+
+            return todoItem;
+          });
+        });
+        setIsEditing(false);
+      })
+      .catch(() => onErrorMesssageChange('Unable to update a todo'))
+      .finally(() => setIsLoading(false));
+  };
+
+  const handleUpdateStatus = () => {
+    handleUpdate({ completed: !todo.completed });
+  };
+
+  const updateTitle = () => {
+    if (editedTitle.trim() === todo.title) {
```

Comment:
You use `editedTitle.trim()` three times, it's better to put it in a variable and reuse it.